### PR TITLE
Fix solids GDML to avoid ORANGE discrepancy

### PR DESCRIPTION
**Pull request by sethrj**
I had previously added a reflected trd to the `solids.gdml`, naming it to sneakily conflict with Geant4's auto-renamed logical volume, but I didn't check the extents thoroughly. This made it look like ORANGE was failing:
trd-old

Renaming the two logical volumes to avoid the collision, and fixing the separation, we find no problem:
trd-new

This highlights future need to have a "consistent" failure mode between ORANGE and VecGeom.

---

Additionally, to reduce less-important "edge case" differences between the geometries, I adjusted a few of the solids not to have internal edges along the trace path (polycone, polyhedra, genpocone).

### CODE REVIEW

File: test/geocel/GeoTests.cc [hunk 1]
```diff
@@ -1729,23 +1729,70 @@ void SolidsGeoTest::test_accessors() const
...
 void SolidsGeoTest::test_trace() const
 {
+    bool const single_orange
+        = (CELERITAS_REAL_TYPE == CELERITAS_REAL_TYPE_FLOAT)
+          && (test_->geometry_type() == "ORANGE");
+
+    if (single_orange)
+    {
+        // TODO: find source of distance misestimation at x=125 (between
+        // boolean and
```
Comment (elliottbiondo) [inline test/geocel/GeoTests.cc]: ...

**Review by elliottbiondo — APPROVED** [review]

**Comment by sethrj** [comment]
FWIW the raytrace error seems to be related to the polyhedron...

solids-bool-poly-single

**Comment by elliottbiondo** [comment]
@sethrj even after adjusting the polyhedron, there are still ray trace failures on develop?

**Comment by sethrj** [comment]
Only with single precision, and I'd say "unexpected" rather than "failures". I updated the test to record the current behavior and we can worry about it later.
